core/templates/dev/head/domain/collection/CollectionObjectFactory.js

```diff
@@ -203,6 +227,83 @@ oppia.factory('CollectionObjectFactory', [
       return skillList;
     };
 
+    // Gets a new ID for a skill. This should be of the same form as in the
+    // backend. This increments nextSkillId.
+    Collection.prototype.getNewSkillId = function() {
+      var newId = 'skill' + String(this._nextSkillId);
+      this._nextSkillId++;
+      return newId;
+    };
+
+    // Adds a new frontend collection skill domain object to this collection.
+    // This will return true if the skill was successfully added, or false if
+    // the given collection skill has an ID already used by another skill.
+    // Changes to the provided object will be reflected in this collection.
+    Collection.prototype.addCollectionSkill = function(collectionSkillObject) {
+      if (!this._skills.hasOwnProperty(collectionSkillObject.getId())) {
+        this._skills[collectionSkillObject.getId()] = collectionSkillObject;
+        return true;
+      }
+      return false;
+    };
+
+    // Attempts to remove a collection skill from this collection given the
+    // specified skill ID. Returns whether the collection skill was
+    // removed, which depends on whether any collection skills reference the
+    // given skill ID.
+    Collection.prototype.deleteCollectionSkill = function(skillId) {
+      if (this._skills.hasOwnProperty(skillId)) {
+        delete this._skills[skillId];
+        this._nodes.forEach(function(node) {
+          if (node.getPrerequisiteSkillList().containsSkill(skillId)) {
+            node.getPrerequisiteSkillList().removeSkillById(skillId);
+          }
+          if (node.getAcquiredSkillList().containsSkill(skillId)) {
+            node.getAcquiredSkillList().removeSkillById(skillId);
+          }
+        });
+        return true;
+      }
+      return false;
+    };
+
+    // Deletes all collection skills within this collection.
+    Collection.prototype.clearCollectionSkills = function() {
```

Comment:
It doesn't look like this is being kept in sync with the skill list object factory. Ditto elsewhere for new skill functionality.